### PR TITLE
Add github action:labeler that labels SQL CLI PRs

### PR DESCRIPTION
**Pull request by pankajkoti**
# Description
## What is the current behavior?

We need to manually label SQL CLI PRs that help us for tracking the releases for SQL CLI

## What is the new behavior?

We are exploring the Github action `labeler` to label Github PR including files affected in the
SQL CLI directory.
https://github.com/actions/labeler

## Does this introduce a breaking change?
No

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: .github/workflows/pr-labeler.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: "Pull Request Labeler"
+on:
+- pull_request_target
```
Comment (pankajkoti) [inline .github/workflows/pr-labeler.yml:3]:
```suggestion
- pull_request
```

File: sql-cli/sql_cli/connections.py [hunk 2]
```diff
@@ -27,6 +27,7 @@ def validate_connections(connections: list[Connection], connection_id: str | Non
             rprint(f"Validating connection {connection.conn_id:{CONNECTION_ID_OUTPUT_STRING_WIDTH}}", status)
 
 
+# dummy comment
```
Comment (pankajkoti) [inline sql-cli/sql_cli/connections.py:30]:
```suggestion
# dummy comment

```

File: sql-cli/sql_cli/connections.py [hunk 3]
```diff
@@ -27,6 +27,8 @@ def validate_connections(connections: list[Connection], connection_id: str | Non
...
+
```
Comment (pankajkoti) [inline sql-cli/sql_cli/connections.py:31]:
```suggestion
```

File: .github/workflows/pr-labeler.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+jobs:
+  triage:
+    permissions:
+      contents: read
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/labeler@v4
+      with:
+        repo-token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (pankajkoti) [inline .github/workflows/pr-labeler.yml:14]:
```suggestion
        repo-token: "${{ secrets.GITHUB_TOKEN }}"
        sync-labels: true
```

File: .github/workflows/pr-labeler.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+name: "Pull Request Labeler"
```
Comment (kaxil) [inline .github/workflows/pr-labeler.yml:1]: Btw there is a bot that I wrote for this and other boooring activities: https://github.com/kaxil/boring-cyborg which could be used as well. We use that in Airflwo and tensorflow

Comment (pankajkoti) [reply .github/workflows/pr-labeler.yml:1]: okay nice, will try adding this.

Comment (pankajkoti) [reply .github/workflows/pr-labeler.yml:1]: I tried installing the app. But looks like it sent a request for approval to be added to Astronomer. Maybe we need to wait until it gets approved.

**Comment by feluelle** [comment]
Great stuff @pankajkoti!

**Review by tatiana — APPROVED** [review]
Really useful!!!

**Pull request merged**